graylog2-server/src/main/java/org/graylog2/inputs/transports/netty/EnvelopeMessageHandler.java

```diff
@@ -27,22 +27,31 @@
 import org.slf4j.LoggerFactory;
 
 import java.net.InetSocketAddress;
+import java.nio.charset.Charset;
 
 public class EnvelopeMessageHandler extends SimpleChannelInboundHandler<AddressedEnvelope<ByteBuf, InetSocketAddress>> {
     private static final Logger LOG = LoggerFactory.getLogger(NettyTransport.class);
+    private static final Charset CHARSET = Charset.defaultCharset();
 
     private final MessageInput input;
+    private final boolean trim;
 
-    public EnvelopeMessageHandler(MessageInput input) {
+    public EnvelopeMessageHandler(MessageInput input, boolean trim) {
         this.input = input;
+        this.trim = trim;
     }
 
     @Override
     protected void channelRead0(ChannelHandlerContext ctx, AddressedEnvelope<ByteBuf, InetSocketAddress> envelope) throws Exception {
         final ByteBuf msg = envelope.content();
         final byte[] bytes = new byte[msg.readableBytes()];
         msg.readBytes(bytes);
-        final RawMessage raw = new RawMessage(bytes, envelope.sender());
+        final RawMessage raw;
+        if (trim) {
+            raw = new RawMessage(new String(bytes, CHARSET).trim().getBytes(CHARSET), envelope.sender());
```

Comment:
This breaks if the payload is not a string. This generic UDP handler shouldn't have string-specific code.

I think we should fix the issue in the Syslog codec instead. For Syslog we can probably fix this without adding an option. (to be verified)

Reply:
Thanks for catching that @bernd. Now that I think about this, the least intrusive way to fix this might be to just trim in the Syslog Codec right where the delegation to the Fortigate pattern happens. Since those (line breaks) break only that type of message/parsing, I think applying the trim there makes sense. This would probably limit the effect to just fixing the issue and not changing any existing parsing behavior for other Syslog formats. What do you think?

https://github.com/Graylog2/graylog2-server/blob/61ec712f023144e1a2fdb59402e12b5eced0f9d0/graylog2-server/src/main/java/org/graylog2/inputs/codecs/SyslogCodec.java#L140

Reply:
@danotorrey Sounds good! :+1:

Reply:
@bernd Done!

Reply:
Thanks!